File: backend/routers/v1/monitor.go
package v1

import (
	monitorAPI "devops-platform/internal/modules/monitor/api"
	"devops-platform/internal/middleware"

	"github.com/gin-gonic/gin"
)

func registerMonitor(r *gin.RouterGroup) {
	g := r.Group("/monitor")
	queryPermission := middleware.RequirePermission("monitor", "list")
	updatePermission := middleware.RequirePermission("monitor", "update")

	// Prometheus config management
	g.GET("/prometheus", queryPermission, monitorAPI.ListPrometheusConfigs)
	g.GET("/prometheus/:id", queryPermission, monitorAPI.GetPrometheusConfig)
	g.POST("/prometheus", updatePermission,
		middleware.SetAuditOperation("Prometheus 配置保存"),
		monitorAPI.SavePrometheusConfig)
	g.PUT("/prometheus/:id", updatePermission,
		middleware.SetAuditOperation("Prometheus 配置更新"),
		monitorAPI.SavePrometheusConfig)
	g.DELETE("/prometheus/:id", updatePermission,
		middleware.SetAuditOperation("Prometheus 配置删除"),
		monitorAPI.DeletePrometheusConfig)
	g.POST("/prometheus/test", queryPermission, monitorAPI.TestPrometheusConnection)

	// Host metrics
	g.GET("/host/metrics", queryPermission, monitorAPI.QueryHostMetrics)
	g.GET("/host/ports", queryPermission, monitorAPI.QueryPortStatus)

	// Agent management
	g.GET("/agent/status", queryPermission, monitorAPI.QueryAgentStatus)
}

File: frontend/src/api/monitor.js
import request from './request'

export const listPrometheusConfigs = (params) => request.get('/monitor/prometheus', { params })
export const getPrometheusConfig = (id) => request.get(`/monitor/prometheus/${id}`)
export const savePrometheusConfig = (data) => request.post('/monitor/prometheus', data)
export const updatePrometheusConfig = (id, data) => request.put(`/monitor/prometheus/${id}`, data)
export const deletePrometheusConfig = (id) => request.delete(`/monitor/prometheus/${id}`)
export const testPrometheusConnection = (data) => request.post('/monitor/prometheus/test', data)
export const queryHostMetrics = (params) => request.get('/monitor/host/metrics', { params })
export const queryPortStatus = (params) => request.get('/monitor/host/ports', { params })
export const queryAgentStatus = (params) => request.get('/monitor/agent/status', { params })
